basic-parser.js

```diff
@@ -0,0 +1,378 @@
+import { parse } from './parser/dsl.js'
+import { readFileSync } from 'fs'
+
+/**
+ * @pineapple_define Parser
+ */
+export function InternalTests () {
```

Comment:
According to the Pineapple Manifesto, this is weird to embed in this file 😄 

I'm going to keep this as-is for now.